internal/manifests/targetallocator/adapters/config_to_prom_config.go

```diff
@@ -257,10 +259,31 @@ func AddHTTPSDConfigToPromConfig(prometheus map[interface{}]interface{}, taServi
 	return prometheus, nil
 }
 
+func WithTLSConfig(caFile, certFile, keyFile, taServiceName string) TAOption {
```

Comment:
this is fine for now, but I do want to simplify this in the future with a similar pattern to #3206